helpers/content_test.go

```diff
@@ -117,3 +122,370 @@ func TestTruncateWordsByRune(t *testing.T) {
+			t.Errorf("Test flag: %d was not found amongs set flags:%d; Result: %d", d.testFlag, flags, flags&d.testFlag)
+		}
+	}
+}
+
+func TestGetHTMLRendererAllFlags(t *testing.T) {
+	type data struct {
+		testFlag int
+	}
+
+	allFlags := []data{
+		{blackfriday.HTML_USE_XHTML},
+		{blackfriday.HTML_FOOTNOTE_RETURN_LINKS},
+		{blackfriday.HTML_USE_SMARTYPANTS},
+		{blackfriday.HTML_SMARTYPANTS_ANGLED_QUOTES},
+		{blackfriday.HTML_SMARTYPANTS_FRACTIONS},
+		{blackfriday.HTML_HREF_TARGET_BLANK},
+		{blackfriday.HTML_SMARTYPANTS_DASHES},
+		{blackfriday.HTML_SMARTYPANTS_LATEX_DASHES},
+	}
+	defaultFlags := blackfriday.HTML_USE_XHTML
+	ctx := &RenderingContext{}
+	ctx.Config = ctx.getConfig()
+	ctx.Config.AngledQuotes = true
+	ctx.Config.Fractions = true
+	ctx.Config.HrefTargetBlank = true
+	ctx.Config.LatexDashes = true
+	ctx.Config.PlainIDAnchors = true
+	ctx.Config.SmartDashes = true
+	ctx.Config.Smartypants = true
+	ctx.Config.SourceRelativeLinksEval = true
+	renderer := GetHTMLRenderer(defaultFlags, ctx)
+	actualFlags := renderer.GetFlags()
+	var expectedFlags int
+	//OR-ing flags together...
+	for _, d := range allFlags {
+		expectedFlags |= d.testFlag
+	}
+	if expectedFlags != actualFlags {
+		t.Errorf("Expected flags (%d) did not equal actual (%d) flags.", expectedFlags, actualFlags)
+	}
+}
+
+func TestGetHTMLRendererAnchors(t *testing.T) {
+	ctx := &RenderingContext{}
+	ctx.DocumentID = "testid"
+	ctx.Config = ctx.getConfig()
+	ctx.Config.PlainIDAnchors = false
+
+	actualRenderer := GetHTMLRenderer(0, ctx)
+	headerBuffer := &bytes.Buffer{}
+	footnoteBuffer := &bytes.Buffer{}
+	expectedFootnoteHref := []byte("href=\"#fn:testid:href\"")
+	expectedHeaderID := []byte("<h1 id=\"id:testid\"></h1>\n")
+
+	actualRenderer.Header(headerBuffer, func() bool { return true }, 1, "id")
+	actualRenderer.FootnoteRef(footnoteBuffer, []byte("href"), 1)
+
+	if !bytes.Contains(footnoteBuffer.Bytes(), expectedFootnoteHref) {
+		t.Errorf("Footnote anchor prefix not applied. Actual:%s Expected:%s", footnoteBuffer.Bytes(), expectedFootnoteHref)
+	}
+
+	if !bytes.Equal(headerBuffer.Bytes(), expectedHeaderID) {
+		t.Errorf("Header Id Postfix not applied. Actual:%s Expected:%s", headerBuffer.Bytes(), expectedHeaderID)
+	}
+}
+
+func TestGetMmarkHtmlRenderer(t *testing.T) {
+	ctx := &RenderingContext{}
+	ctx.DocumentID = "testid"
+	ctx.Config = ctx.getConfig()
+	ctx.Config.PlainIDAnchors = false
+	actualRenderer := GetMmarkHtmlRenderer(0, ctx)
+
+	headerBuffer := &bytes.Buffer{}
+	footnoteBuffer := &bytes.Buffer{}
+	expectedFootnoteHref := []byte("href=\"#fn:testid:href\"")
+	expectedHeaderID := []byte("<h1 id=\"id\"></h1>")
+
+	actualRenderer.FootnoteRef(footnoteBuffer, []byte("href"), 1)
+	actualRenderer.Header(headerBuffer, func() bool { return true }, 1, "id")
+
+	if !bytes.Contains(footnoteBuffer.Bytes(), expectedFootnoteHref) {
+		t.Errorf("Footnote anchor prefix not applied. Actual:%s Expected:%s", footnoteBuffer.Bytes(), expectedFootnoteHref)
```

Comment:
I think buff. String() is what the user wants to see printed here ... or use something other thant %s.